### DISCUSSION TITLE
Future plans for __x.$data property and potential deprecation

### DISCUSSION THREAD

**Discussion by gaslit** [alpinejs/alpine#1194] [1. Help]
Hello all. Massive kudos to Caleb, Hugo and the AlpineJS team and community for a magnificent Javascript library that is well designed.

One simple and yet pressing question:
https://codewithhugo.com/alpinejs-inspect-component-data-from-js/

With regards to the above, I have been using this __x.$data undocumented hack for a long time and I am wondering if

1. There is a better way of accessing out of scope data properties in pure Javascript?

2. Are there any risks of it being deprecated in any upcoming updates?

Many thanks.

**Comment by HugoDF** [comment]
1. There's a `$component` helper at https://github.com/alpine-collective/alpine-magic-helpers (which uses the same `$data` approach under the hood)

2. Will have to wait for @calebporzio to demo/release v3, I don't think there are plans to change anything in v2, I'm sure even in v3 there'll be something similar (at the very least so that devtools can access/read state)

Side note: Alpine is Caleb's project, I only write random blog posts 😉.

**Reply by northern-mindset** [reply]
2. Doesn't seem so: #1543

**Reply by HugoDF** [reply]
There's `_x_dataStack` although it's now an array so accessing data is `el._x_dataStack[0]` although like that discussion says it's probably an anti pattern now
